packages/theme/src/cli/utilities/theme-environment/local-assets.ts

```diff
@@ -87,64 +92,160 @@ function findLocalFile(event: H3Event, ctx: DevServerContext) {
   )
 }
 
-function isCompiledAssetRequest(event: H3Event) {
-  return event.path.includes('/compiled_assets')
+function isCompiledAssetRequest(event: H3Event): boolean {
+  const pathname = parseServerEvent(event).pathname
+
+  if (!pathname.includes('/compiled_assets')) {
+    return false
+  }
+
+  const assetPath = pathname.split('/').at(-1)
+
+  return (
+    assetPath === 'styles.css' ||
+    assetPath === 'block-scripts.js' ||
+    assetPath === 'snippet-scripts.js' ||
+    assetPath === 'section-scripts.js'
+  )
 }
 
 function handleCompiledAssetRequest(event: H3Event, ctx: DevServerContext) {
-  const assetPath = parseServerEvent(event).pathname.split('/').at(-1)
+  if (!isCompiledAssetRequest(event)) return
+
+  const pathname = parseServerEvent(event).pathname
+  const assetPath = pathname.split('/').at(-1)
+
+  // isKnownCompiledAssetType already guarantees assetPath is defined.
+  if (!assetPath) return
+
+  switch (assetPath) {
+    case 'styles.css':
+      return handleStylesCss(ctx, event)
+    case 'block-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'block')
+    case 'snippet-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'snippet')
+    case 'section-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'section')
```

Comment:
I think this one is called just `scripts.js`? It currently goes to the proxy instead of being generated locally:

<img width="653" alt="image" src="https://github.com/user-attachments/assets/2707ed95-4db9-4e5c-a9a4-ceacf2229854" />